### PR TITLE
Feature/upload deployment package s3

### PR DESCRIPTION
**Pull request by agduncan94**
https://github.com/dockstore/dockstore/issues/3324

This adds a circleci config which validates the JS files and uploads to S3 if valid.
Uploads a deployment zip for lambda.

I only included the index.js files, I'm not sure if the other files (ex. YML files) are required.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                type: string
+            s3BucketFolder:
+                type: string
+        steps:
+            - run:
+                name: Validate index.js
+                command: node --check << parameters.lambdaFolder >>/index.js
+            - run:
+                name: Create deployment zip
+                command: zip << parameters.lambdaFolder >>/function.zip << parameters.lambdaFolder >>/index.js
```
Comment (coverbeck) [inline .circleci/config.yml:50]: Cool, did not know you could pass parameters like that.

Comment (agduncan94) [reply .circleci/config.yml:50]: Yeah, it is a pretty cool feature.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        steps:
+            - when:
+                condition: <<parameters.aws_bucket>>
+                steps:
+                    - checkout
+                    - node/install:
+                        install-yarn: false
+                        node-version: latest
+                    - create_zip_upload_to_s3:
+                        lambdaFolder: "cloud-watch-to-slack-testing-58006136-c30b-43d1-880c-7b36f860e7c8/deployment"
```
Comment (Ldcabansay) [inline .circleci/config.yml:22]: jw why are all these folder names so weird?

Comment (agduncan94) [reply .circleci/config.yml:22]: Not entirely sure, I think they have relevance on AWS.

**Pull request merged**